sdk/src/types/block/address/bech32.rs

```diff
@@ -0,0 +1,93 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use core::str::FromStr;
+use std::hash::Hash;
+
+use serde::{de::Visitor, Deserialize, Deserializer, Serialize, Serializer};
+
+use crate::types::block::address::Address;
+
+/// An address and its network type.
+#[derive(Clone, Eq, PartialEq, Hash, Serialize, Deserialize)]
+pub struct Bech32Address {
+    pub(crate) inner: Address,
+    #[serde(rename = "bech32Hrp")]
+    pub(crate) hrp: String,
+}
+
+impl AsRef<Address> for Bech32Address {
+    fn as_ref(&self) -> &Address {
+        &self.inner
+    }
+}
+
+impl FromStr for Bech32Address {
+    type Err = crate::wallet::Error;
+
+    fn from_str(address: &str) -> Result<Self, Self::Err> {
+        let (address, hrp) = Address::try_from_bech32_with_hrp(address)?;
+
+        Ok(Self::new(address, hrp))
+    }
+}
+
+impl Bech32Address {
+    /// Create a new address wrapper.
+    pub fn new(address: Address, hrp: String) -> Self {
+        Self { inner: address, hrp }
+    }
+
+    /// Get the bech32 human readable part
+    pub fn hrp(&self) -> &str {
+        &self.hrp
+    }
```

Comment:
Also add address() getter?

Reply:
it has as_ref impl, dont think thats enough?

Reply:
I think it's not so obvious that it's available this way